spec/controllers/posts_controller_spec.rb

```diff
@@ -31,7 +31,7 @@
       end
     end
 
-    describe "GET send_for_purchase" do
+    describe "POST send_for_purchase" do
```

Comment:
The route is defined as post so just clearing this up over here

`post "/posts/:id/send_for_purchase/:purchase_id", to: "posts#send_for_purchase", as: :send_for_purchase`